### PR TITLE
Basecamp React component

### PR DESCRIPTION
**Pull request by mcasper**
A React component for the basecamp integration.

Further work: The todos should be grouped by the todolist they belong to. There's already a method on the Basecamp::Client for this, but my limited React knowledge kept me from getting the front end part to work. (Collections inside of collections got too hard for me)

### CODE REVIEW

File: app/assets/javascripts/Components/Basecamp/BasecampStore.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  load() {
+    var BasecampTodosCollection = Backbone.Collection.extend({
+      url: window.location.pathname + "/basecamp/todos"
+    });
+
+    var collection = new BasecampTodosCollection();
+
+    collection.fetch({
+      success: (collection, _response, _options) => {
```
Comment (mcasper) [inline app/assets/javascripts/Components/Basecamp/BasecampStore.js]: >.<

Comment (mikeastock) [inline app/assets/javascripts/Components/Basecamp/BasecampStore.js]: Probably should leave them without the underscore so we are consistent?

File: app/assets/javascripts/Components/Basecamp/BasecampTodolist.js.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+var BasecampActions = require("./BasecampActions");
+var BasecampTodo = require("./BasecampTodo.js.jsx");
+
+var BasecampTodolist = React.createClass({
+  getInitialState() {
+    debugger;
```
Comment (mikeastock) [inline app/assets/javascripts/Components/Basecamp/BasecampTodolist.js.jsx]: Oops left a debugger

File: app/assets/javascripts/Components/Basecamp/BasecampTodolist.js.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+var BasecampActions = require("./BasecampActions");
+var BasecampTodo = require("./BasecampTodo.js.jsx");
+
+var BasecampTodolist = React.createClass({
```
Comment (mikeastock) [inline app/assets/javascripts/Components/Basecamp/BasecampTodolist.js.jsx]: This component isn't used anywhere. We should probably remove it.

**Pull request merged**